### ISSUE TITLE
IndexError during data processing

### ISSUE THREAD

**Issue by bnic24** [deepset-ai/FARM#48]
While putting together a notebook to demonstrate another issue I encountered the following error:

```
/content/FARM/farm/data_handler/utils.py in read_docs_from_txt(filename, delimiter, encoding)
    137 
    138         # if last row in file is not empty
--> 139         if all_docs[-1] != doc:
    140             all_docs.append({"doc": doc})
    141             # sample_to_doc.pop()

IndexError: list index out of range
```
The error should be reproducible by running the following colab notebook: https://colab.research.google.com/drive/1BWW207df5kHoYrXQOkEwDw5zVv_SnwP-

Perhaps it's just some sort of data cleaning issue, but maybe it's worth looking into if only to  make data processing more robust.

**Comment by tholor** [comment]
Hey bnic24, 

I had a quick look and I think the reason is that your train.txt contains only a single "doc" with 1000 sentences. You will need multiple docs separated by empty lines.

I suppose this is due to this cell in the Colab Notebook: 
```
with open('dataset.txt', 'r') as file, open('train.txt', 'w') as train_file, open('dev.txt', 'w') as dev_file, open('test.txt', 'w') as test_file:
  for i, line in enumerate(file):
    # create a doc every 1000 lines, for illustration purposes only
    if i > 0 and i % 1000 == 0:
      line = line + '\n'
    # create train/dev/test sets
    if i < 1000:
      train_file.write(line)
    elif i < 19500:
      dev_file.write(line)
    else:
      test_file.write(line)
``` 
It worked for me after changing the splits to 10k and 15k:  
```
with open('dataset.txt', 'r') as file, open('train.txt', 'w') as train_file, open('dev.txt', 'w') as dev_file, open('test.txt', 'w') as test_file:
  for i, line in enumerate(file):
    # create a doc every 1000 lines, for illustration purposes only
    if i > 0 and i % 1000 == 0:
      line = line + '\n'
    # create train/dev/test sets
    if i < 10000:
      train_file.write(line)
    elif i < 15000:
      dev_file.write(line)
    else:
      test_file.write(line)
```
However, I totally agree that this can be confusing for the user and we should add a warning / more descriptive exception. If you have time to add this, happy to see a PR :). The right part of the code would be here: 
https://github.com/deepset-ai/FARM/blob/ded5f4612ae83d4f3b6fbbf440b4a2eb8d914280/farm/data_handler/utils.py#L139-L143
If you don't have time, no worries. I can also take care of this.

**Comment by bnic24** [comment]
Hey, tholor,

Thanks for looking into this! I see the problem. I don't have time right now, but I might later this week, or early next week. If someone hasn't dealt with it by then, I'll have a go.

**Comment by tholor** [comment]
Fixed by #69
